developer/src/tike/xml/layoutbuilder/builder.js

```diff
@@ -18,8 +18,8 @@ $(function() {
   this.saveSelection = function() {
     let key = builder.selectedKey(), subKey = builder.selectedSubKey();
     return {
-      id: key ? $(key).data('id') : null,
-      subId: subKey ? $(subKey).data('id') : null
+      id: key.length ? $(key).data('id') : null,
```

Comment:
Did this previously check if key or subKey are null?

Reply:
Yeah it turns out that `builder.selectedKey()` cannot return null -- the previous check was wrong. jquery returns an empty array (with its own extra props patched in) if there is no selected key.

jquery patterns are not ideal in this editor. But they are what we have